bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/Response.java

```diff
@@ -0,0 +1,384 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.mideaac.internal.handler.CommandBase.FanSpeed;
+import org.openhab.binding.mideaac.internal.handler.CommandBase.OperationalMode;
+import org.openhab.binding.mideaac.internal.handler.CommandBase.SwingMode;
+import org.openhab.binding.mideaac.internal.handler.Timer.TimerData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Response} performs the byte data stream decoding
+ * The original reference is
+ * https://github.com/georgezhao2010/midea_ac_lan/blob/06fc4b582a012bbbfd6bd5942c92034270eca0eb/custom_components/midea_ac_lan/midea/devices/ac/message.py#L418
+ *
+ * @author Jacek Dobrowolski - Initial contribution
+ * @author Bob Eckhoff - Add Java Docs, minor fixes
+ */
+@NonNullByDefault
+public class Response {
+    byte[] data;
+
+    // set empty to match the return from an empty byte avoid null
+    float empty = (float) -19.0;
+    private Logger logger = LoggerFactory.getLogger(Response.class);
+
+    private final int version;
+    String responseType;
+    byte bodyType;
+
+    private int getVersion() {
+        return version;
+    }
+
+    /**
+     * Response class Parameters
+     * 
+     * @param data byte array from device
+     * @param version version of the device
+     * @param responseType response type
+     * @param bodyType Body type
+     */
+    public Response(byte[] data, int version, String responseType, byte bodyType) {
+        this.data = data;
+        this.version = version;
+        this.bodyType = bodyType;
+        this.responseType = responseType;
+
+        logger.debug("Power State: {}", getPowerState());
+        logger.trace("Prompt Tone: {}", getPromptTone());
+        logger.trace("Appliance Error: {}", getApplianceError());
+        logger.debug("Target Temperature: {}", getTargetTemperature());
+        logger.debug("Operational Mode: {}", getOperationalMode());
+        logger.debug("Fan Speed: {}", getFanSpeed());
+        logger.debug("On Timer: {}", getOnTimer());
+        logger.debug("Off Timer: {}", getOffTimer());
+        logger.debug("Swing Mode: {}", getSwingMode());
+        logger.trace("Auxiliary Heat: {}", getAuxHeat());
+        logger.trace("Eco Mode: {}", getEcoMode());
+        logger.debug("Sleep Function: {}", getSleepFunction());
+        logger.debug("Turbo Mode: {}", getTurboMode());
+        logger.trace("Fahrenheit: {}", getFahrenheit());
+        logger.debug("Indoor Temperature: {}", getIndoorTemperature());
+        logger.debug("Outdoor Temperature: {}", getOutdoorTemperature());
+        logger.debug("LED Display: {}", getDisplayOn());
+        logger.trace("Humidity: {}", getHumidity());
+        logger.trace("Alternate Target Temperature {}", getAlternateTargetTemperature());
```

Comment:
Do we really not to log all this? If so i would suggest to surround the debug or trace parts with a condition: `logger.isTraceEnabled`  or `logger.isDebugEnabled`

Reply:
Ok Done.  I guess this saves a little I/O when not logging at that level?

Reply:
Yes exactly